lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -329,32 +330,28 @@ def run_sync(command, stdin = nil)
       stdout = ''
       stderr = ''
       exit_status = nil
-      started = false
 
-      channel = session.open_channel do |ch|
-        ch.on_data do |c, data|
-          stdout.concat(data)
-        end
-        ch.on_extended_data { |_, _, data| stderr.concat(data) }
-        ch.on_request('exit-status') { |_, data| exit_status = data.read_long }
-        # Send data to stdin if we have some
-        ch.send_data(stdin) unless stdin.nil?
-        # on signal: sending the signal value (such as 'TERM')
-        ch.on_request('exit-signal') do |_, data|
-          exit_status = data.read_string
-          ch.close
-          ch.wait
-        end
-        ch.exec command do |_, success|
-          raise 'could not execute command' unless success
-
-          started = true
+      master, slave = PTY.open
+      read, write = IO.pipe
+      pid = spawn('/usr/bin/ssh', @ssh_user + '@' + @host, command, :in => read, :out => slave)
+      read.close
+      slave.close
+
+      unless stdin.nil?
+        write.puts(stdin)
+        write.close
+        loop do
+          ret = begin
+                  master.gets
+                rescue Errno::EIO
+                  nil
+                end
```

Comment:
The exception gets raised only when you attempt to read after the source has reached eof. Since you're looping anyway, you could start checking for `#eof?` instead of catching the exception.

Reply:
This still applies